File: src/scala/ComputeArea.scala
package scala
import javafiles.Student

/**
 * Created by sushi on 15-06-03.
 */
object ComputeAreaScala extends App {
  
  val radius = 5
  val PI = 3.14159
  val area = radius * radius * PI
  println ("the radius is " + area)

  val i = 0
  val p  = 0
  val k  = 100.0
  val j  = i + 1
  System.out.println("j is " + j + " and k is " + k + " p is " + p)

  // WE will call a Java class from a scala class.
  // THe java class is called Student

  val sushi = new Student(12,"Norman")
  println(sushi.getStudentId)
  println(sushi.getStudentName)
  sushi.setStudentId(13)
  println(sushi.getStudentId)

  // declaring a scala student class which will used by our java class.
  case class ScalaStudent(studentId: Int,studentName: String)
}

File: src/javafiles/ComputeArea.java
package javafiles;

import scala.ComputeAreaScala;

/**
 * Created by sushi on 15-06-03.
 */
public class ComputeArea {

    public static void main(String[] args) {
        double radius = 5;
        final double PI = 3.14159;
        double area = radius * radius * PI;
        System.out.println("the area is " + area);

        int i = 0;
        int p = 0;
        double k = 100.0;
        int j = i + 1;
        System.out.println("j is " + j + " and k is " + k + "p is " + p);

        // Here we will utilize a Scala object from a Java Class
        // The scala student class is a case class.
        // lets use the scala student class
        ComputeAreaScala.ScalaStudent scalaStudent = new ComputeAreaScala.ScalaStudent(13, "SushiScala");
        System.out.println(scalaStudent.studentId());
        System.out.println(scalaStudent.studentName());
    }
}